pep-0596.rst

```diff
@@ -0,0 +1,191 @@
+
+For example, "Command Prompt" uses legacy code page by default.
+But WSL changes the code page to 65001, and  ``python.exe`` on Windows
+can be executed from WSL.  So ``python.exe`` executed from legacy
+console and from WSL cannot read text files written by each other.
+
+Many Windows users don't understand code pages and which code page is
+currently used.  So changing default text file encoding based on current
+code page may cause confusion.
+
+Consistent default text encoding will make Python behavior more expectable
+and easy to learn.
+
+
+Use UTF-8 by default is easier to new programmers
+-------------------------------------------------
+
+Python is one of the most popular first programming language.
+
+New programmers may not know about encoding.  When they download text data
+written in UTF-8 from the internet, they are forced to know encoding.
+
+Popular text editers like VS Code or Atom use UTF-8 by default.
+Even notepad.exe uses UTF-8 by default from Windows 10 2019 may update.
+(Note that Python 3.9 will be released in 2021.)
+
+It would be nice if new programmers are not forced to know about encoding
+until they need to handle text files encoded in encoding other than UTF-8.
+
+
+Specification
+=============
+
+From Python 3.9, default encoding of ``TextIOWrapper`` and ``open()`` is
+changed from ``locale.getpreferredencoding(False)`` to "UTF-8".
+
+When there is device encoding (``os.device_encoding(buffer.fileno())``),
+itstill precedes than this default encoding.
+
+
+Not affected areas
+------------------
+
+Unlike UTF-8 mode, ``locale.getpreferredencoding(False)`` returns
+locale encoding.  Locale encoding is still respected when it makes sense.
+
+``stdin``, ``stdout``, and ``stderr`` keep respecting locale too.  For example,
+these commands don't cause mojibake regardless code page::
+
+   > python -c "print('こんにちは')" | more
+   こんにちは
+   > python -c "print('こんにちは')" > temp.txt
+   > type temp.txt
+   こんにちは
+
+Pipes and TTY should use locale encoding:
+
+* ``subprocess`` and ``os.popen`` use locale encoding because subprocess
+  will use locale encoding.
+* ``getpass.getpass`` use locale encoding when using TTY.
+
+Of course, this PEP is about file content, not file path.  So filenames
+or file paths are not affected.
+
+
+Affected APIs
+--------------
+
+All code using default encoding of ``TextIOWrapper`` or ``open`` are
+affected.  This is incomplete list of APIs affected by this PEP:
+
+* ``lzma.open``, ``gzip.open``, ``bz2.open``, ``ZipFile.read_text``.
+* ``socket.makefile``
+* ``tempfile.TemporaryFile``, ``tempfile.NamedTemporaryFile``
+* ``trace.CoverageResults.write_results_file``
+
+
+Warning
+-------
+
+From 3.8, ``DeprecationWarning`` is shown when encoding is omitted and
+``locale.getpreferredencoding(False)`` is not UTF-8.  This helps not only
+writing forward compatible code, but also investigating unexpected
+``UnicodeDecodeError`` caused by the default encoding.  (See Motivation section
+above.)
```

Comment:
DeprecationWarning means that the feature (open without encoding) will be removed soon? What about my proposed open(encoding="locale") idea?

Reply:
No.  It is not removed, it just changes behavior.

See https://docs.python.org/3/library/warnings.html#warning-categories

> Previously DeprecationWarning and FutureWarning were distinguished based on whether a feature was being removed entirely or changing its behaviour. They are now distinguished based on their intended audience and the way they’re handled by the default warnings filters.

DeprecationWarning is Warning for Python developers.  It can be used to notice
behavior changes.